### ISSUE TITLE
Hooks are called twice if the provider is aliased

### ISSUE THREAD

**Issue by lafeuil** [nestjs/nest#3182]
## Current behavior
Hooks are called twice if the provider is aliased.

## Input Code
In a sample NestJS application (`$ nest new`), implement the `OnModuleInit` hook in the `AppService`
```ts
@Injectable()
export class AppService implements OnModuleInit {
  private onModuleInitCount = 0;
  onModuleInit() {
    if (this.onModuleInitCount++ > 0) {
      throw new Error('onModuleInitCount has been called twice');
    }
  }
  getHello(): string {
    return 'Hello World!';
  }
}
```
Add an alias to `AppService` in `AppModule`
```ts
@Module({
  imports: [],
  controllers: [AppController],
  providers: [
    AppService,
    {
      provide: 'AliasedAppService',
      useExisting: AppService,
    },
  ],
})
export class AppModule {}
```

## Expected behavior
I think, each hook may be called once by instance.

**Comment by peawyoyoyin** [comment]
Hello. I looked into the code and found a root cause. I will open a PR with the changes to fix this.

**Comment by kamilmysliwiec** [comment]
Fixed in 6.10.13
